cmd/onboard.go

```diff
@@ -152,123 +151,152 @@ func doOnboard() error {
 		CipherSuite:          kexCipherSuiteID,
 		AllowCredentialReuse: true,
 	})
-	if err != nil {
-		return err
-	}
 	if newDC == nil {
-		fmt.Println("Credential not updated (Credential Reuse Protocol")
+		slog.Info("Credential not updated")
 		return nil
 	}
 
 	// Store new credential
-	fmt.Println("FIDO Device Onboard Complete")
+	slog.Info("FIDO Device Onboard Complete")
 	return updateCred(*newDC, FDO_STATE_IDLE)
 }
 
-func transferOwnership(ctx context.Context, rvInfo [][]protocol.RvInstruction, conf fdo.TO2Config) (*fdo.DeviceCredential, error) { //nolint:gocyclo
-	var to2URLs []string
-	directives := protocol.ParseDeviceRvInfo(rvInfo)
-	for _, directive := range directives {
-		if !directive.Bypass {
-			continue
-		}
-		for _, url := range directive.URLs {
-			to2URLs = append(to2URLs, url.String())
-		}
-	}
-
-	// Try TO1 on each address only once
+func transferOwnership(ctx context.Context, rvInfo [][]protocol.RvInstruction, conf fdo.TO2Config) *fdo.DeviceCredential { //nolint:gocyclo
 	var (
-		to1d     *cose.Sign1[protocol.To1d, []byte]
-		to1dErrs []error
+		directives          = protocol.ParseDeviceRvInfo(rvInfo)
+		onboardingPerformed bool
+		rvEntryDelay        time.Duration
+		newDC               *fdo.DeviceCredential
 	)
-TO1:
-	for _, directive := range directives {
-		if directive.Bypass {
-			continue
-		}
-
-		for _, url := range directive.URLs {
-			var err error
-			to1d, err = fdo.TO1(context.TODO(), tls.TlsTransport(url.String(), nil, insecureTLS), conf.Cred, conf.Key, nil)
-			if err != nil {
-				slog.Error("TO1 failed", "base URL", url.String(), "error", err)
-				to1dErrs = append(to1dErrs, err)
-				continue
-			}
-			break TO1
-		}
 
-		if directive.Delay != 0 {
-			// A 25% plus or minus jitter is allowed by spec
-			select {
-			case <-ctx.Done():
-				return nil, ctx.Err()
-			case <-time.After(directive.Delay):
-			}
-		}
-	}
-	if to1d == nil {
-		return nil, errors.Join(to1dErrs...)
+	if len(directives) == 0 {
+		slog.Error("No rendezvous directives found in device credential")
+		return nil
```

Comment:
in rust we explicitely bail https://github.com/fdo-rs/fido-device-onboard-rs/blob/main/client-linuxapp/src/main.rs#L310 we should do the same I guess